saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -38,6 +44,15 @@ class Meta:
         error_type_class = ProductError
         error_type_field = "product_errors"
 
+    @classmethod
+    def bulk_action(cls, queryset):
+        products = models.Product.objects.none()
+        for category in queryset:
+            products = products | unpublished_products_before_category_delete(category)
+        product_ids = [product.pk for product in products]
```

Comment:
A bit heavy, but we can leave it like that for now. Though it would be nice to add a test in the benchmark package even though with https://github.com/mirumee/saleor/issues/4359 it's still not very useful.